src/tools/wasm2c.cc

```diff
@@ -119,12 +119,19 @@ static void ParseOptions(int argc, char** argv) {
 
 // TODO(binji): copied from binary-writer-spec.cc, probably should share.
 static std::string_view strip_extension(std::string_view s) {
-  std::string_view ext = s.substr(s.find_last_of('.'));
-  std::string_view result = s;
-
-  if (ext == ".c")
-    result.remove_suffix(ext.length());
-  return result;
+  auto pos = s.find_last_of('.');
+  if (pos != std::string_view::npos) {
+    std::string_view ext = s.substr(pos);
+    std::string_view result = s;
+
+    if (ext == ".c") {
+      result.remove_suffix(ext.length());
+      return result;
+    }
+  }
+  fprintf(stderr, "output filename must end in .c");
+  exit(1);
```

Comment:
How about if we instead just return early if not `.` is found?

That would seem to fix better with the name of the extension, and avoid having to create this new error condition.

Reply:
In fact I think you can maybe just remove this function completely and use `wabt::StripExtension`

Reply:
(...what if we rename it to "base name" and make it a breaking change?)

Reply:
IIUC `basename` normally means stripping the directory component, not the extension component.

Reply:
we mean the output flag, `-o`, we could make it take a "base name": `-o empty.c` -> `-o empty`. which makes somewhat more sense since it also generates a `.h`.

Reply:
I think if we just strip the extension that is fine.   I think it makes sense to write `-o foo.c` (at least to me).

Reply:
(i.e no need to do a breaking change here I think).